File: frontend/src/api/messenger.ts
import { apiJson } from './client'

export type ChatChannel = {
  id: string
  title: string
  project_id: string | null
  task_id: string | null
  created_by_user_id: string
  created_at: string
  message_count: number
}

export type ChatMessage = {
  id: string
  channel_id: string
  user_id: string
  author_email: string
  text: string
  created_at: string
}

export function listChatChannels(token: string): Promise<ChatChannel[]> {
  return apiJson<ChatChannel[]>('/messenger/channels', token)
}

export function createChatChannel(
  token: string,
  payload: { title: string; project_id?: string; task_id?: string },
): Promise<ChatChannel> {
  return apiJson<ChatChannel>('/messenger/channels', token, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  })
}

export function listChatMessages(token: string, channelId: string): Promise<ChatMessage[]> {
  return apiJson<ChatMessage[]>(`/messenger/channels/${channelId}/messages`, token)
}

export function postChatMessage(token: string, channelId: string, text: string): Promise<ChatMessage> {
  return apiJson<ChatMessage>(`/messenger/channels/${channelId}/messages`, token, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ text }),
  })
}

File: backend/app/schemas/messenger.py
"""Схемы мессенджера."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ChatChannelCreate(BaseModel):
    title: str = Field(min_length=2, max_length=200)
    project_id: UUID | None = None
    task_id: UUID | None = None


class ChatChannelRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    project_id: UUID | None
    task_id: UUID | None
    created_by_user_id: UUID
    created_at: datetime
    message_count: int = 0


class ChatMessageCreate(BaseModel):
    text: str = Field(min_length=1, max_length=10_000)


class ChatMessageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    channel_id: UUID
    user_id: UUID
    author_email: str
    text: str
    created_at: datetime
